dask_geopandas/hilbert_distance.py

```diff
@@ -43,7 +43,9 @@ def _hilbert_distance(gdf, total_bounds=None, level=16):
     # Compute distance along hilbert curve
     distances = _encode(level, x, y)
 
-    return pd.Series(distances, index=gdf.index, name="hilbert_distance")
+    return pd.Series(
+        distances, index=gdf.index, name="hilbert_distance", dtype=np.int64
```

Comment:
Wouldn't `distances` already be a `uint32`? I suspect that the changes will need to be at a lower level (maybe in `_encode`), and then when we're constructing a Series here the dtype will just be the dtype of whatever ndarray is passed.

Reply:
Yes you are right, i had missed that this is actually an unsigned int. The reason for the overflow is actually that the value is greater than the max value for int but not uint, and dask simply isn't respecting the dtype.